docs/manual/src/proc_macro/functions.md

```diff
@@ -59,11 +63,18 @@ impl TextSplitter {
 }
 ```
 
-Supported default values:
+Supported defaults without a literal:
 
-  - String, integer, float, and boolean literals
-  - `[]` for empty Vecs
-  - `Option<T>` allows either `None` or `Some(T)`
+- All `Option` types default to `None`, Vecs and Maps default to empty items
+- All builtin types have default values (zero, false, empty)
+- Named types (ie, Records, Objects, etc) are supported, but you must ensure the
+  item is able to be created without arguments (eg, has a default constructor
+  with no args, is a record with all fields having defaults, etc)
```

Comment:
Looks right to me, but I think describing 1 type per row would be clearer.  It's less compact, but I think it's worth the space.  Maybe a table would be better:

| Type | Default value | 
|---------|----------------------|
| `Option` | `None` |
| String | Empty string |
| `Vec<T>` | Empty list |
| `HashMap<T>` | Empty map |
| Builtin numeric types | 0 |
| `bool` | false |
| Records | Record with all fields set to their default value (only valid if they all have defaults) |
| Objects | Primary constructor called with 0 arguments |

Reply:
thanks - that just makes the section below this look even worse, but I just opened #2551 instead of getting distracted by that.